### PR TITLE
Add new rule for warning against use of weak types

### PR DESCRIPTION
**Pull request by cloudkite**
Warns when using weak type annotations **any**, **Object** and **Function**

### CODE REVIEW

File: src/rules/noWeakTypes.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import {get} from 'lodash';
```
Comment (gajus) [inline src/rules/noWeakTypes.js]: To keep inline with the rest of the codebase, please change this to:

```js
import _ from 'lodash';
```

We are using https://github.com/lodash/babel-plugin-lodash to rewire them in the build step.

**Comment by gajus** [comment]
Thank you!

**Comment by cloudkite** [comment]
👍 cheers!

**Comment by danharper** [comment]
@gajus the latest version published to npm was 2.11.4

**Comment by gajus** [comment]
@danharper Released version 2.14.0.

Looks like CI script is silently failing. I will investigate why.

**Comment by danharper** [comment]
👍 Thanks

**Comment by gajus** [comment]
@danharper Release script fixed and tested across multiple repositories.

**Comment by Couto** [comment]
This is just a question so that newbies like me can have some clarification:

If `Function` is a weak type (no argue there) and should be avoided, what's the proper way of documenting something like a compose function (per example)? 

`const compose = (f:Function, g:Function):Function => (x:mixed): mixed => f(g(x));` 

I'm new to flow and types, so please bear with me. :)

**Comment by cloudkite** [comment]
@Couto you'll probably get more responses over at https://github.com/facebook/flow.

I would approach this with a polymorphic function see https://flowtype.org/docs/functions.html#polymorphic-functions

```
type Func<T, R> = (t: T) => R 

function compose<In, Out, T>(f: Func<T, Out>, g: Func<In, T>): Func<T , Out>  {
   return x => f(g(x))
}
```

* you could do it without the `Func` type alias. added for clarity
* could be a arrow function. but find full functions are a bit easier to annotation especially when its a polymorphic functions

**Pull request merged**